File: USACO_Jan2024/BalancingBacteria.py
def solve(a, ans, start):
    # power is = to length
    power = len(a) - start
    # multiplicity = first element
    multiplicity = a[start]
    ans[0] += abs(multiplicity)
    if multiplicity == 0:
        return

    for i in range(len(a) - 1, start, -1):
        a[i] += (power * multiplicity * -1)
        power -= 1



N = int(input())


a = list(map(int, input().split()))

ans = [0]
for i in range(N):
    solve(a, ans, i)

print(ans[0])

File: USACO_Jan2024/BalancingBacteria.cpp
#include<bits/stdc++.h>
using namespace std;

typedef long long ll;

void solve(vector<ll>& a, int& ans) {
    // power is = to length
    int power = a.size();
    // multiplicity = first element
    ll multiplicity = a[0]; ans += abs(multiplicity);
    if (multiplicity == 0) return;

    for (int i = a.size() - 1; i > 0; i--) {
        a[i] += (power * multiplicity * -1);
        power --;
    }
}

int main() {
    // inputing N number of patches of field
    int N; cin >> N;

    // input each path
    vector<ll> a(N);
    for (int i = 0; i < N; i++) cin >> a[i];

    // call function to find the answer
    int ans = 0;
    // for the length of the vector
    for (int i = 0; i < N; i++) {
        solve(a, ans);

        a.erase(a.begin());
    }

    cout << ans << endl;

}
